### PR TITLE
Split Host and Component management

### PR DESCRIPTION
**Pull request by chrisrohr**
* Component management is on its own page now
* Components per host is still viewable on the host page
* Add ability to update host and component
* Add auth to ensure calls that require admin are protected

### CODE REVIEW

File: service/src/main/java/org/kiwiproject/champagne/App.java [hunk 1]
```diff
@@ -15,28 +13,11 @@
...
-import org.kiwiproject.champagne.dao.ComponentDao;
-import org.kiwiproject.champagne.dao.DeployableSystemDao;
-import org.kiwiproject.champagne.dao.DeploymentEnvironmentDao;
-import org.kiwiproject.champagne.dao.HostDao;
-import org.kiwiproject.champagne.dao.ReleaseDao;
-import org.kiwiproject.champagne.dao.ReleaseStatusDao;
-import org.kiwiproject.champagne.dao.TaskDao;
-import org.kiwiproject.champagne.dao.TaskStatusDao;
-import org.kiwiproject.champagne.dao.UserDao;
+import org.kiwiproject.champagne.dao.*;
```
Comment (sleberknight) [inline service/src/main/java/org/kiwiproject/champagne/App.java:16]: Assume something in your IDE config got reset to cause star imports?

File: service/src/main/java/org/kiwiproject/champagne/dao/DeployableSystemDao.java [hunk 2]
```diff
@@ -47,9 +47,6 @@ public interface DeployableSystemDao {
     @SqlQuery("select system_admin from users_deployable_systems where user_id = :userId and deployable_system_id = :systemId")
     boolean isUserAdminOfSystem(@Bind("userId") long userId, @Bind("systemId") long systemId);
 
-    @SqlQuery("select true from users_deployable_systems uds join users u on u.id = uds.user_id join deployable_systems ds on ds.id = uds.deployable_system_id where u.system_identifier = :userName and ds.id = :systemId")
```
Comment (sleberknight) [inline service/src/main/java/org/kiwiproject/champagne/dao/DeployableSystemDao.java:50]: What was this used for?

File: service/src/main/java/org/kiwiproject/champagne/dao/HostDao.java [hunk 3]
```diff
@@ -22,6 +22,9 @@ public interface HostDao {
     @GetGeneratedKeys
     long insertHost(@BindBean Host host, @Bind("tagCsv") String tagCsv);
 
+    @SqlUpdate("update hosts set hostname = :hostname, tags = :tags where id = :id")
+    int updateHost(@Bind("hostname") String hostname, @Bind("tags") String tagCsv, @Bind("id") long id);
```
Comment (sleberknight) [inline service/src/main/java/org/kiwiproject/champagne/dao/HostDao.java:26]: Longer term (or maybe now), would it be better to use an array type in the database? Pros include more semantically correct and maybe easier to query, cons might include less portability across DBs. And more I have not thought about.

File: service/src/main/java/org/kiwiproject/champagne/model/DeployableSystemThreadLocal.java [hunk 4]
```diff
@@ -1,20 +1,29 @@
...
 
+import java.util.Optional;
+
 @UtilityClass
 public class DeployableSystemThreadLocal {
 
-    private static final ThreadLocal<Long> THREAD_LOCAL = new ThreadLocal<>();
+    @Getter
+    @AllArgsConstructor
+    public class DeployableSystemInfo {
```
Comment (sleberknight) [inline service/src/main/java/org/kiwiproject/champagne/model/DeployableSystemThreadLocal.java:14]: Would this ever need to be compared, stringified? If so could use `@Value`

File: service/src/main/java/org/kiwiproject/champagne/resource/DeploymentEnvironmentResource.java [hunk 5]
```diff
@@ -16,16 +11,15 @@
...
 import javax.validation.Valid;
-import javax.ws.rs.Consumes;
-import javax.ws.rs.DELETE;
-import javax.ws.rs.GET;
-import javax.ws.rs.POST;
-import javax.ws.rs.PUT;
-import javax.ws.rs.Path;
-import javax.ws.rs.PathParam;
-import javax.ws.rs.Produces;
+import javax.ws.rs.*;
```
Comment (sleberknight) [inline service/src/main/java/org/kiwiproject/champagne/resource/DeploymentEnvironmentResource.java:14]: Ditto star import...won't repeat after this

File: service/src/test/java/org/kiwiproject/champagne/resource/filter/DeployableSystemResponseFilterTest.java [hunk 6]
```diff
@@ -1,21 +1,21 @@
...
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+
 @DisplayName("DeployableSystemResponseFilter")
 class DeployableSystemResponseFilterTest {
 
     @Test
     void shouldClearDeployableSystem() {
-        DeployableSystemThreadLocal.setCurrentDeployableSystem(1L);
+        DeployableSystemThreadLocal.setCurrentDeployableSystem(1L, false);
```
Comment (sleberknight) [inline service/src/test/java/org/kiwiproject/champagne/resource/filter/DeployableSystemResponseFilterTest.java:18]: After seeing all the `boolean` arguments, would it be better to use an `enum` for the user type? e.g. something like:

```java
DeployableSystemThreadLocal.setCurrentDeployableSystem(1L, UserType.ADMIN);
```

File: service/src/main/java/org/kiwiproject/champagne/dao/DeployableSystemDao.java (hunk 2 above)
Comment (chrisrohr) [reply service/src/main/java/org/kiwiproject/champagne/dao/DeployableSystemDao.java:50]: It was used to check if a user was in a system based on their username not just their id. Not needed anymore

File: service/src/main/java/org/kiwiproject/champagne/dao/HostDao.java (hunk 3 above)
Comment (chrisrohr) [reply service/src/main/java/org/kiwiproject/champagne/dao/HostDao.java:26]: I'll create a ticket to look into switching it out.

File: service/src/main/java/org/kiwiproject/champagne/model/DeployableSystemThreadLocal.java (hunk 4 above)
Comment (chrisrohr) [reply service/src/main/java/org/kiwiproject/champagne/model/DeployableSystemThreadLocal.java:14]: At this time no

File: service/src/test/java/org/kiwiproject/champagne/resource/filter/DeployableSystemResponseFilterTest.java (hunk 6 above)
Comment (chrisrohr) [reply service/src/test/java/org/kiwiproject/champagne/resource/filter/DeployableSystemResponseFilterTest.java:18]: I know you hate boolean parameters. I'll create an issue to relook at this construct.

**Review by sleberknight — APPROVED** [review]

**Pull request merged**